### PR TITLE
Allow configuration of nginx ingress controller

### PR DESCRIPTION
**Pull request by mueller-ma**
# Description

Allow configuration of default nginx certificate.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [ ] Small minor change not affecting the Ansible Role code (GitHub Actions Workflow, Documentation etc.)

## How Has This Been Tested?

I executed the role on my cluster and set `rke2_ingress_nginx_default_certificate: "cert-manager/default-cert"`. With cert-manager I created a certificate in the secret `cert-manager/default-cert`.

### CODE REVIEW

File: tasks/ingress-nginx.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+- name: Copy ingress-nginx files to first server
+  ansible.builtin.template:
+    src: "templates/ingress-nginx-config.yml.j2"
+    dest: "{{ rke2_data_path }}/server/manifests/rke2-ingress-nginx-config.yaml"
```
Comment (MonolithProjects) [inline tasks/ingress-nginx.yml:4]: Make sure the destination directory exist. It seems like it is not. See https://github.com/lablabs/ansible-role-rke2/actions/runs/9078504653/job/25340508968

File: tasks/ingress-nginx.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+    owner: root
+    group: root
+    mode: 0664
```
Comment (MonolithProjects) [inline tasks/ingress-nginx.yml:7]: Please add new line at end of the line. See https://github.com/lablabs/ansible-role-rke2/actions/runs/9078504650/job/25340508973?pr=216

**Review by MonolithProjects — CHANGES_REQUESTED** [review]

File: tasks/ingress-nginx.yml (hunk 1 above)
Comment (mueller-ma) [reply tasks/ingress-nginx.yml:13]: Done

File: tasks/ingress-nginx.yml (hunk 2 above)
Comment (mueller-ma) [reply tasks/ingress-nginx.yml]: Done. Also created https://github.com/lablabs/ansible-role-rke2/pull/222 to fix this issue automatically.

**Comment by mueller-ma** [comment]
I updated the PR to make all values in this file configurable.

**Review by MonolithProjects — APPROVED** [review]
LGTM

**Comment by elbvdk** [comment]
I think this should be marked as a Breaking Change; if you previously used the custom manifest deployment method to deploy those nginx configuration options, this will overwrite that custom manifest. 

In our case, our deployment was failing because use-forwarded-hearders: true was removed from the nginx config which caused our Rancher to get stuck in an HTTP 308 redirect loop.

**Comment by MonolithProjects** [comment]
That's a good point @elbvdk. Thanks. Will add that to the release notes

**Comment by MonolithProjects** [comment]
@elbvdk if you want to keep using custom manifest for nginx config, use release [1.34.1](https://github.com/lablabs/ansible-role-rke2/releases/tag/1.34.1) where this issue is fixed (`rke2_ingress_nginx_values` will be used only if it's not empty)

**Comment by elbvdk** [comment]
Thanks for the quick fix!

**Pull request merged**